src/main/java/strategy/PurchaseOperation.java

```diff
@@ -0,0 +1,10 @@
+package strategy;
+
+import model.FruitTransaction;
+
+public class PurchaseOperation implements OperationHandler {
+    @Override
+    public int getQuantityToAdd(FruitTransaction fruitTransaction) {
+        return -fruitTransaction.getQuantity();
```

Comment:
According to the checklist, 'Handling Purchase operation': You should check the result balance before saving it in the Storage - it should be positive. Throw `RuntimeException` in case the balance is negative. Right now you're just returning a negative quantity without any checks.

Reply:
fix comment

Reply:
Not resolved